### DISCUSSION TITLE
How do I detect objects on video from code to get bounding boxes and quantity of objects? Can you please share a simple working example.

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [ultralytics/yolov5#9920] [Q&A]
How do I detect objects on video from code to get bounding boxes and quantity of objects? 
I haven't find a working example. Is there a simple working example of detection a video maybe even from youtube?

How do I skip frame or do something else to improve performance?

**Comment by glenn-jocher** [comment]
@martinenkoEduard YouTube and Video inference examples are shown in README:

<img width="877" alt="Screenshot 2022-10-25 at 17 14 19" src="https://user-images.githubusercontent.com/26833433/197812974-12fa07d0-88da-4bb4-b375-9632035a5187.png">

**Reply by martinenkoEduard** [reply]
Yep, but how do I get objects (amount, bounding boxes and probability)  programmatically  for each frame?

**Reply by glenn-jocher** [reply]
@martinenkoEduard 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://github.com/ultralytics/yolov5/issues/251) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://github.com/ultralytics/yolov5/issues/36) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by spacewalk01** [comment]
Here is an example:
```
import torch
import cv2
import os
from os.path import join

VIDEO_PATH = "your video path"
model_path = "yolov5s.pt" # your model path

# Model
model = torch.hub.load('ultralytics/yolov5', 'custom', path=model_path, force_reload=True)  

SCORE_THRESH = 0.4

cap = cv2.VideoCapture(VIDEO_PATH)

while True:
    ret, image = cap.read()
    i += 1    
        
    if not ret:
        break
    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
    
    results = model(image)
    df = results.pandas().xyxy[0]

    df = df[df.confidence > SCORE_THRESH]
    
    if len(df) > 0:
        # Bounding box
        start_point = (int(df.xmin[0]), int(df.ymin[0]))
        end_point = (int(df.xmax[0]), int(df.ymax[0]))
        image = cv2.rectangle(image, start_point, end_point, (0, 0, 255), 5) 
        
       	cv2.imshow('video',image)
	cv2.waitKey(0)

cap.release()
```
